girder/api/v1/file.py

```diff
@@ -322,3 +323,23 @@ def updateFileContents(self, file, params):
         .param('size', 'Size in bytes of the new file.', dataType='integer')
         .notes('After calling this, send the chunks just like you would with a '
                'normal file upload.'))
+
+    @access.user
+    @loadmodel(model='file', level=AccessType.READ)
+    def copy(self, file, params):
+        user = self.getCurrentUser()
+        fileModel = self.model('file')
+        item = None
+
+        if 'itemId' in params:
+            item = self.model('item').load(params['itemId'], user=user,
+                                           level=AccessType.WRITE, exc=True)
+
+        newFile = fileModel.copyFile(file, user, item=item)
+
+        return fileModel.filter(newFile, user)
+
+    copy.description = (
+        Description('Copy a file.')
+        .param('id', 'The ID of the file.', paramType='path')
+        .param('itemId', 'The item to copy the file to.', required=False))
```

Comment:
For this endpoint, I vote that we always require an itemId parameter since the behavior of copying a file into the same item is probably going to be very uncommon. It will also simplify the method, since we could use

    @loadmodel(model='file', level=AccessType.READ)
    @loadmodel(model='item', map={'itemId': 'item'}, level=AccessType.WRITE)
    def copy(self, file, item, params):